src/browser/ui/ReactDOMComponent.js

```diff
@@ -232,6 +234,24 @@ function processChildContext(context, tagName) {
   return context;
 }
 
+var onlyNestTextTags = {
+  'option': true
+};
+
+function getChildrenTextContent(tag, children) {
+  var childrenContent = '';
+  ReactChildren.forEach(children, function (c) {
+    if ("production" !== process.env.NODE_ENV) {
+      ("production" !== process.env.NODE_ENV ? warning(
+        !ReactElement.isValidElement(c),
+        tag + ' can not have nested tags'
+      ) : null);
+    }
+    childrenContent += c;
```

Comment:
I feel like maybe we should skip the piece of child content if it is not a `string` or `number`.  Maybe something like `if(typeof c !== 'string' && typeof c !== 'number') { warning(false, '%s can not have complex children/content', tag); } else { childrenContent += c; }`, instead of checking for a valid element as the warning condition.